### PR TITLE
Update docs and some scripts cleanup

### PR DESCRIPTION
**Pull request by alsora**

### CODE REVIEW

File: cross-compiling/README.md [hunk 1]
```diff
@@ -27,153 +38,93 @@ The supported architectures are:
...
+ - The cross-compilation script will mount the workspace as a Docker volume. This does not go well with symbolic links. For this reason ensure that the workspace contains the whole source code and not a symlink to the repositories.
+ - It is recommended that the workspace contains the source code of all the ROS 2 dependencies of your packages. However, you can also cross-compile individual packages if you already have the cross-compiled dependencies, see the [advanced section](advanced.md).
 
-Ignore some packages and python dependencies (note that there is a different script for each distribution you want to  cross-compile)
+We provide convenient scripts for downloading the ROS 2 sources for a specific distribution and for ignoring some packages that are not meant to be cross-compiled, e.g. visualization tools.
+If you want to cross-compile generic ROS 2 packages, see the [advanced section](advanced.md).
 
-    cd -
-    bash ignore_pkgs.sh ~/ros2_cc_ws eloquent
+    bash get_ros2_sources.sh --distro=foxy --ros2-path=~/ros2_cc_ws
```
Comment (mauropasse) [inline cross-compiling/README.md:90]: I think if you run this command, a directory `~` will be created in the same directory as the script `get_ros2_sources.sh`
You could do 
```
bash get_ros2_sources.sh --distro=foxy --ros2-path=$HOME/ros2_cc_ws
```

File: cross-compiling/README.md [hunk 2]
```diff
@@ -27,153 +38,93 @@ The supported architectures are:
...
-
-One of the goals of this cross-compilation framework is to be quite easy to be extended to new target architectures.
-
-For example, if you want to add a target named `my_architecture` you need the following:
-
- - A compilation environment for your architecture. You have to create a new Dockerfile named `docker_environments/Dockerfile_my_architecture`. This Dockerfile must begin with the line
-    ```
-    FROM ros2_cc_base
-    ```
+If you want access toe ROS 2 command line tools, you will have to source the cross-compiled workspace.
```
Comment (mauropasse) [inline cross-compiling/README.md:117]: minor typo to fix if you add another commit

File: cross-compiling/README.md [hunk 3]
```diff
@@ -27,153 +38,93 @@ The supported architectures are:
...
+ros2 run examples_rclcpp_minimal_publisher publisher_lambda
+```
 
 ## References
 
- - [ROS2 cross-compilation official tools](https://github.com/ros2/cross_compile)
- - [ROS2 step by step cross-compilation on ARM](https://github.com/ros2-for-arm/ros2/wiki/ROS2-on-arm-architecture)
- - [ROS2 RaspberryPi cross-compilation](https://github.com/alsora/ros2-raspberrypi)
+ - [ROS 2 cross-compilation official tools](https://github.com/ros2/cross_compile)
+ - [ROS 2 step by step cross-compilation on ARM](https://github.com/ros2-for-arm/ros2/wiki/ROS 2-on-arm-architecture)
```
Comment (mauropasse) [inline cross-compiling/README.md:129]: The link has now a space `ROS\ 2`

File: cross-compiling/advanced.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+This readme contains some more advanced cross-compilation features provided by this framework.
+
+
+## Add new ROS 2 packages to be cross-compiled
+
+You should always first start from a standard ROS 2 workspace, e.g. obtained from
+
+```
+bash get_ros2_sources.sh --distro=foxy --ros2-path=~/ros2_cc_ws
```
Comment (mauropasse) [inline cross-compiling/advanced.md:11]: Same as previous comment, `bash get_ros2_sources.sh --distro=foxy --ros2-path=$HOME/ros2_cc_ws`

File: cross-compiling/advanced.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+To start a build, **from within the Docker environment**
+```
+bash /root/compilation_scripts/cross_compile.sh
+```
+
+You can modify the cross_compile script `/root/compilation_scripts/cross_compile.sh` **from within the Docker environment** to customize your debugging.
+Common changes are the following:
+
+ - Comment the line that clears the workspace every time to speed up builds `rm -rf build install log`
+ - Build only the package that is failing, adding one of the following to the `colcon build` arguments: `--packages-up-to MY_PACKAGE` or `--packages-select MY_PACKAGE`.
```
Comment (mauropasse) [inline cross-compiling/advanced.md:113]: I use a lot also 
```
--packages-start <PKG UNTIL THE COMPILATION WAS SUCCESFUL>
```

File: cross-compiling/advanced.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+If for any reason you don't want to cross-compile a whole ROS 2 workspace made of all the ROS 2 core packages, but you only want to cross-compile a specific individual package, you can do that by first cross-compiling the standard ROS 2 SDK, as described in the [README](README.md), and then you can copy the cross-compiled libraries into the sysroot of your target architecture.
+
+```
+cp -R ~/ros2_cc_ws/install/cmake sysroots/<TARGET_ARCHITECTURE>/usr/cmake
+cp -R ~/ros2_cc_ws/install/share/* sysroots/<TARGET_ARCHITECTURE>/usr/share/
+cp -R ~/ros2_cc_ws/install/lib/* sysroots/<TARGET_ARCHITECTURE>/usr/lib/
+cp -R ~/ros2_cc_ws/install/include/* sysroots/<TARGET_ARCHITECTURE>/usr/include/
+```
+
+**NOTE:** calling `bash get_sysroot.sh` will erase the current sysroot and generate a new one from scratches, thus losing the copied ROS 2 SDK.
```
Comment (mauropasse) [inline cross-compiling/advanced.md:127]: We should do same as in `get_ros2_sources.sh`, i.e. warn that the directory already exsist. The user should remove it first manually

File: cross-compiling/README.md (hunk 1 above)
Comment (alsora) [reply cross-compiling/README.md:90]: Do you know why this happens?
`~/ros2_cc_ws` is actually the default...

File: cross-compiling/advanced.md (hunk 6 above)
Comment (alsora) [reply cross-compiling/advanced.md:127]: Yes, agreed

File: cross-compiling/README.md (hunk 1 above)
Comment (alsora) [reply cross-compiling/README.md:90]: Fixed in
https://github.com/irobot-ros/ros2-performance/pull/52/commits/15b57f7e418298684be7e4df387a54c25e048f48

**Pull request merged**